File: ParkHere/UI/parkhere_desktop/lib/model/parking_session.dart
import 'package:json_annotation/json_annotation.dart';
import 'parking_reservation.dart';

part 'parking_session.g.dart';

@JsonSerializable()
class ParkingSession {
  final int id;
  final int parkingReservationId;
  final DateTime? actualStartTime;
  final DateTime? arrivalTime;
  final DateTime? actualEndTime;
  final int? extraMinutes;
  final double? extraCharge;
  final DateTime createdAt;
  final ParkingReservation parkingReservation;

  ParkingSession({
    this.id = 0,
    this.parkingReservationId = 0,
    this.actualStartTime,
    this.arrivalTime,
    this.actualEndTime,
    this.extraMinutes,
    this.extraCharge,
    required this.createdAt,
    required this.parkingReservation,
  });

  factory ParkingSession.fromJson(Map<String, dynamic> json) => _$ParkingSessionFromJson(json);
  Map<String, dynamic> toJson() => _$ParkingSessionToJson(this);
}

File: ParkHere/ParkHere.Services/Database/ParkingSession.cs
using ParkHere.Services.Database;
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace ParkHere.Services.Database
{
    public class ParkingSession
    {
        [Key]
        public int Id { get; set; }

        [Required]
        public int ParkingReservationId { get; set; }

        public DateTime? ActualStartTime { get; set; }
        public DateTime? ArrivalTime { get; set; }
        public DateTime? ActualEndTime { get; set; }

        public int? ExtraMinutes { get; set; }
        public decimal? ExtraCharge { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.Now;

        [ForeignKey(nameof(ParkingReservationId))]
        public ParkingReservation ParkingReservation { get; set; } = null!;
    }
}